flake.nix

```diff
@@ -130,64 +131,51 @@
 
           modules = haskellModules system;
         };
-    in
-    {
+
       project = perSystem projectFor;
       flake = perSystem (system: (projectFor system).flake { });
+    in
+    flake-parts.lib.mkFlake { inherit inputs; } {
+      imports = [
+        ./pre-commit.nix
+      ];
+      flake = {
+        inherit project flake;
+
+        # This will build all of the project's executables and the tests
+        check = perSystem
+          (system:
+            (nixpkgsFor system).runCommand "combined-check"
+              {
+                nativeBuildInputs = builtins.attrValues self.checks.${system}
+                  ++ builtins.attrValues self.flake.${system}.packages;
+              } ''mkdir $out''
+          );
+
+        haskellModules = perSystem haskellModules;
+
+        # Instruction for the Hercules CI to build on x86_64-linux only, to avoid errors about systems without agents.
+        herculesCI.ciSystems = [ "x86_64-linux" ];
```

Comment:
Extract into hercules-ci.nix file and import. Configure flake-updates and see https://github.com/mlabs-haskell/lambda-buffers/blob/617ac9202cd9e3bc67ff85bdc3d7e2fd6ec2d13a/hercules-ci.nix as an example.

If Plutip has cool docs, you might want to also add the Github pages support. Not really urgent.